src/coverlet.core/DependencyInjection.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+
+using Coverlet.Core.Abstracts;
+using Coverlet.Core.Helpers;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Coverlet.Core
+{
+    internal static class DependencyInjection
+    {
+        private static Lazy<IServiceProvider> _serviceProvider = new Lazy<IServiceProvider>(() => InitDefaultServices(), true);
+
+        public static IServiceProvider Current
+        {
+            get
+            {
+                return _serviceProvider.Value;
+            }
+        }
+
+        public static void Set(IServiceProvider serviceProvider)
```

Comment:
this method needed to "plug new services collection" used by testing to pass mocked IRetryHelper aware of process exit and file locking.